packages/block-library/src/index.js

```diff
@@ -65,14 +65,17 @@ import * as classic from './classic';
 /**
  * Function to register core blocks provided by the block editor.
  *
+ * @param {boolean} enableLegacyWidgets     Optional. Editor setting to enable Legacy Widget Block.
+ * @param {boolean} enableMenu              Optional. Editor setting to enable Menu Block.
+ *
  * @example
  * ```js
  * import { registerCoreBlocks } from '@wordpress/block-library';
  *
  * registerCoreBlocks();
  * ```
  */
-export const registerCoreBlocks = () => {
+export const registerCoreBlocks = ( enableLegacyWidgets, enableMenu ) => {
```

Comment:
With this change, we will ship all this code in the next WordPress major release. Shouldn't we keep `process.env.GUTENBERG_PHASE === 2` so users don't get these blocks bundled at all?

This is also a public API change as we introduce 2 new params. They are temporary by nature so I don't think this is something that we want to support forever in the current shape. We should seek to find an alternative approach of how we register those blocks we still consider as unstable. Have you considered exposing another experimental method which registers blocks which are guarded by those feature flags?

Reply:
That's a much better idea, thanks @gziolo ! Fixed now.